File: DSA/binaryTree.c
#include<stdio.h>
#include<stdlib.h>

typedef struct  nodetype{
    struct nodetype * left;
    int num;
    struct nodetype * right;

}node;
node * memory(){
    node * temp;
    temp= (node*) malloc(sizeof(node));
    temp->left=0;
    temp->right=0;
    return temp;
}
node * create(){
    node * new_node=memory();
    int n;
    printf("Enter the data and -1 to exit");
    scanf("%d",&n);
    if (n==-1)
    return NULL;
    new_node->num=n;
    printf("Enter left child of %d\n",n);
    new_node->left=create();    
    printf("Enter right child of %d\n",n);
    new_node->right=create();
    return new_node;

}
void inorder(node * root){
    if (root == NULL)
        return;
inorder(root->left);
printf("%d ", root->num);
inorder(root->right); 
}

void main(){
    node * root;
    root=create();
    inorder(root);
}

File: DSA/Java/BinaryTree.java
public class BinaryTree {
    static class Node{
        int data;
        Node left;
        Node right;
        Node(int data){
            this.data=data;
            this.left=null;
            this.right=null;
        }
    }
    static class CreateBinaryTree{
        static int idx=-1;
        public static Node BuildTree(int []nodes){
             idx++;
             if (nodes[idx]==-1) {
                return null;
             }
             Node newNode= new Node(nodes[idx]);
             newNode.left =BuildTree(nodes);
             newNode.right=BuildTree(nodes);
             return newNode;
        }
    }
    public static void main(String[] args) {
        int []nodes={1,2,4,-1,-1,3,-1,-1,5,6,-1,55};
        CreateBinaryTree cbt=new CreateBinaryTree();
        Node root=cbt.BuildTree(nodes);

    }
}
